warrior/WarriorCore/exec_type_driver.py

```diff
@@ -11,65 +11,349 @@
+    """
+    operator = operator.lower()
+    if operator == "ge":
+        return True if get_object_from_datarepository(exec_condition) >= exec_cond_var else False
+    elif operator == "gt":
+        return True if get_object_from_datarepository(exec_condition) > exec_cond_var else False
+    elif operator == "le":
+        return True if get_object_from_datarepository(exec_condition) <= exec_cond_var else False
+    elif operator == "lt":
+        return True if get_object_from_datarepository(exec_condition) < exec_cond_var else False
+
+    pNote("Unknown error occur when deciding value, please check condition value of the step",
+          "Error")
+    return False
+
+def logical_decision(exec_condition, exec_cond_var, operator="equal"):
+    """
+        Handle the logical decision for the value comparison
+        :param:
+            exec_condition: value from data_repo to be compared
+            exec_cond_var: user provided value to be compared
+            operator: math operator in plain English
+        :return:
+            True if condition match, else return False
+    """
+    status = True
+    result = None
+    if type(get_object_from_datarepository(exec_condition)) != type(exec_cond_var):
+        pNote("Comparing different type of value, please check the conditional value type", "ERROR")
+        status = False
+    elif operator in ["ge", "gt", "le", "lt"] and\
+         not isinstance(exec_condition, int) and not isinstance(exec_condition, float):
+        pNote("Comparing non-numerical value using numerical operator,"\
+            "please check value and operator type", "ERROR")
+        status = False
+
+    if status and operator == "eq":
+        result = True if get_object_from_datarepository(exec_condition) == exec_cond_var else False
+    elif status and operator == "ne":
+        result = True if get_object_from_datarepository(exec_condition) != exec_cond_var else False
+    elif status and operator in ["ge", "gt", "le", "lt"]:
+        result = math_decision(exec_condition, exec_cond_var, operator)
+    else:
+        pNote("Execution condition failed for expected value: {} , operator: {}, actual value: {}"\
+              .format(exec_cond_var, operator,
+                      get_object_from_datarepository(exec_condition)), "WARNING")
+
+    return result
+
+def rule_parser(rule):
+    """
+        Parse an xml rule element and call logical decision function
+        :param:
+            rule: xml element for a single rule
+        :return:
+            status of the rule
+    """
+    exec_condition = rule.get("Condition", None)
+    exec_cond_var = rule.get("Condvalue", None)
+    else_action = rule.get("Else", None)
+    if else_action is not None and else_action.upper() == "GOTO":
+        else_action = rule.get("Elsevalue")
+
+    # Check for math operator
+    support_operators = ["ge", "gt", "le",
+                         "lt", "eq", "ne"]
+    operator = rule.get("Operator", "eq")
```

Comment:
can we have a global var with this list instead of duplicating this list in most of the functions here? I think the dictionary that I mentioned in math_decision can be taken globally and then used in different functions appropriately.

Reply:
I will put this in #245

Reply:
Change has been made in
https://github.com/warriorframework/warriorframework/pull/245